### PR TITLE
New structural application elements

### PR DESCRIPTION
**Pull request by rubenzorrilla**
This PR adds the new truss and shell elements to the structural mechanics application interface. The old beam element has been commented (once the local axes issue is completely addressed we can uncomment it to include the new corotational beam).

Recall that the truss element issue regarding the CROSS_AREA variable, which is required to be written as KratosMultiphysics.StructuralMechanicsApplication.CROSS_AREA, is still pending...

### CODE REVIEW

**Comment by philbucher** [comment]
I tried to load the branch in GiD but I get the following error:
```
error "not well-formed (invalid token)" at line 269 character 8
"ype="Beam">
    <!--h <--Error-- ere we could add a list of all of the p"
error "not well-formed (invalid token)" at line 269 character 8
"ype="Beam">
    <!--h <--Error-- ere we could add a list of all of the p"
    while executing
"dom parse [tDOM::xmlReadFile [file join $dir xml $ElementsFileName]] doc"
    (procedure "Model::getElements" line 6)
    invoked from within
"Model::getElements Elements.xml"
    (procedure "Structural::xml::Init" line 6)
    invoked from within
"Structural::xml::Init"
    (file "/home/philippb/software/GiD/gid13.1.8d-x64/problemtypes/kratos.gid/apps/Structural/xml/GetFromXML.tcl" line 115)
    invoked from within
"::gidutils::source /home/philippb/software/GiD/gid13.1.8d-x64/problemtypes/kratos.gid/apps/Structural/xml/GetFromXML.tcl"
    ("uplevel" body line 1)
    invoked from within
"uplevel 1 ::gidutils::source $args"
    (procedure "source" line 7)
    invoked from within
"source /home/philippb/software/GiD/gid13.1.8d-x64/problemtypes/kratos.gid/apps/Structural/xml/GetFromXML.tcl"
    ("uplevel" body line 1)
    invoked from within
"uplevel #0 [list source [file join $dir xml GetFromXML.tcl]]"
    (procedure "LoadMyFiles" line 4)
    invoked from within
"LoadMyFiles"
    (procedure "::Structural::Init" line 19)
    invoked from within
"::Structural::Init"
    (file "/home/philippb/software/GiD/gid13.1.8d-x64/problemtypes/kratos.gid/apps/Structural/start.tcl" line 44)
    invoked from within
"::gidutils::source /home/philippb/software/GiD/gid13.1.8d-x64/problemtypes/kratos.gid/apps/Structural/start.tcl"
    ("uplevel" body line 1)
    invoked from within
"uplevel 1 ::gidutils::source $args"
    (procedure "source" line 7)
    invoked from within
"source /home/philippb/software/GiD/gid13.1.8d-x64/problemtypes/kratos.gid/apps/Structural/start.tcl"
    ("uplevel" body line 1)
    invoked from within
"uplevel 2 [list source $fileName]"
    (procedure "apps::loadAppFile" line 2)
    invoked from within
"apps::loadAppFile $fileName"
    (class "::App" method "activate" line 6)
    invoked from within
"$app activate"
    (procedure "apps::setActiveApp" line 8)
    invoked from within
"apps::setActiveApp Structural"
    invoked from within
".gid.win_example.information.imgStructural invoke "
    invoked from within
".gid.win_example.information.imgStructural instate !disabled { .gid.win_example.information.imgStructural invoke } "
    invoked from within
".gid.win_example.information.imgStructural instate pressed { .gid.win_example.information.imgStructural state !pressed; .gid.win_example.information.i..."
    (command bound to event)
```

**Comment by rubenzorrilla** [comment]
@philbucher, It was my bad when commenting the old beam element... I've removed it (we have the repo to check it if needed) .

**Comment by philbucher** [comment]
Now it works , looks good!
I couldn't see the linear truss element though, did you not add it?

**Comment by rubenzorrilla** [comment]
Hi @philbucher! Should be... :confused:

File: kratos.gid/apps/Structural/xml/Elements.xml [hunk 1]
```diff
@@ -239,40 +259,69 @@
       <parameter n="PK2_STRESS_TENSOR" pn="PK2 Stress Tensor" v="false" state="hidden"/>
     </outputs>
   </ElementItem>
+
   <!--structural elements-->
   <!--beam elements-->
-  <ElementItem n="BeamElement" pn="Small displacements beam" ImplementedInFile="small_displacement_beam_element_3D2N.cpp" ImplementedInApplication="StructuralMechanicsApplication" MinimumKratosVersion="9000" ProductionReady="Developer" WorkingSpaceDimension="3D" LocalSpaceDimension="1" RequiresLocalAxes="True" LocalAxesAutomaticFunction="Structural::xml::AddLocalAxesToBeamElement" LargeDeformation="False" help="This element implements a small displacement timoshenko beam structural model" RotationDofs="True" ElementType="Beam">
+
+  <!--truss elements-->
+  <ElementItem n="TrussElement3D2N" pn="Large displacements truss" ImplementedInFile="truss_element_3D2N.cpp" ImplementedInApplication="StructuralMechanicsApplication"
```
Comment (philbucher) [inline kratos.gid/apps/Structural/xml/Elements.xml]: This is all I can see, I don't see the `TrussLinearElement3D2N`
Both trusses are implemented in `truss_element_3D2N`, but they are in the Interface as `TrussLinearElement3D2N` and `TrussElement3D2N`

Comment (rubenzorrilla) [reply kratos.gid/apps/Structural/xml/Elements.xml]: Copy & paste error. My bad...

**Comment by philbucher** [comment]
now it is there!

@KlausBSautter and I tried setting up a case. It works except that `CROSS_AREA` is not found bcs it is not written with a proceeding `KratosMultiphysics.StructuralMechanicsApplication`

**Comment by rubenzorrilla** [comment]
@philbucher @KlausBSautter, the unique things that are left to be done are: 

- CROSS_AREA variable writting: @jginternational is already aware of that. Maybe he can come up with a temporary hacky solution. But taking into account that this can happen in other applications, IMO we have to handle it with care.
- Filter elements according to the analysis_type (linear/nonlinear): Correct me if I'm wrong, but IMO the large displacements elements should only appear if the "analysis_type" field is "nonlinear".
- The beam does not appear yet.

I suggest to merge the changes in this PR and open new branches to tackle the issues above.

**Comment by KlausBSautter** [comment]
@rubenzorrilla  I agree that the "analysis_type" should influence the possible choice of elements

**Comment by jginternational** [comment]
@rubenzorrilla Thinking about the Cross area full name, the easy way is to modify ConstitutiveLaws.xml and change n="CROSS_AREA" -> n="KratosMultiphysics.StructuralMechanicsApplication.CROSS_AREA"

This should write the full name, but I do not like this solution so much

**Comment by rubenzorrilla** [comment]
@jginternational, for me it is a not so hacky solution that would allow to run truss cases directly from the interface, thus I would add it as a temporary solution.

**Pull request merged**